Add trimDB logic for expired machines

saalieri opened the pull request:
- Implemented trim method in machines module to perform hard delete of expired machine entries
- Added new `trimDB` CLI command to trigger database cleanup manually or via external schedulers
- Added config variables `AUTO_RUN_TRIM_DB_CMD = True` that toggles use of external cleanup and `RETURNED_VM_TTL` that defines machine record retention period in days before purge
- Added db_name to the configuration
- Updated GCE provider documentation to include configuration setup
- Bump version 0.3.4 to 0.3.5

Fixes #35

- [ /] Tests pass
- [/ ] Appropriate changes to documentation are included in the PR

saalieri committed:
Add purgeMachines command and trim logic for expired machines

- Implemented trim method in machines module to perform hard delete of expired machine entries
- Added new `purgeMachines` CLI command to trigger database cleanup manually or via external schedulers
- Added config variables `SCHEDULER_USE_PURGE_CMD` that toggles use of external cleanup and `DB_RETURN_VM_TTL` that defines machine record retention period in days before purge
- Added db_name to the configuration
- Updated GCE provider documentation to include configuration setup
- Bump version 0.3.4 to 0.3.5

clineberry commented on hf-provider/src/gce_provider/config.py:
```diff
@@ -33,6 +33,8 @@
 DEFAULT_HF_TEMPLATES_FILENAME = "gcpgceinstprov_templates.json"
 DEFAULT_CONFIG_FILENAME = "gcpgceinstprov_config.json"
 DEFAULT_DB_FILENAME = DEFAULT_HF_PROVIDER_NAME
+DEFAULT_SCHEDULER_USE_PURGE_CMD = False
```
See comment from the SYMPHONY_GCE_INSTALL.md

clineberry commented on hf-provider/src/gce_provider/config.py:
```diff
@@ -33,6 +33,8 @@
 DEFAULT_CONFIG_FILENAME = "gcpgceinstprov_config.json"
 DEFAULT_DB_FILENAME = DEFAULT_HF_PROVIDER_NAME
+DEFAULT_SCHEDULER_USE_PURGE_CMD = False
+DEFAULT_RETURN_VM_TTL = "30" # 30 days
```
See comment from the SYMPHONY_GCE_INSTALL.md

clineberry reviewed (changes requested):
A few comments. 

Please ensure @corykim reviews the updates to ../db/machines.py file.

saalieri committed:
Rename and update cleanup-related variables and commands

- Replaced `SCHEDULER_USE_PURGE_CMD = False` with `AUTO_RUN_TRIM_DB_CMD = True`
  and updated all related configuration references.
- Renamed `RETURN_VM_TTL` to `RETURNED_VM_TTL` across codebase and config.
- Refactored cleanup logic in `machines.py` to align with the new config variables.
- Renamed `purgeMachines` function and command to `trimDB`.
- Updated `SYMPHONY_GCE_INSTALL` documentation to reflect new variable names and command changes.

saalieri commented on hf-provider/SYMPHONY_GCE_INSTALL.md:
```diff
@@ -161,8 +161,8 @@ The following configuration variables are supported. Variables that have no defa
 | `PUBSUB_AUTOLAUNCH`     | If set to `true`, the provider will attempt to automatically launch the PubSub event listener. If `false`, you will need to launch the PubSub event listener manually, via the command `hf-monitor`. You can launch the daemon inline with a command, with the command `hf-gce <command> --monitor`. | `true`                                                                       ...
-
-
+| `SCHEDULER_USE_PURGE_CMD`     | If set to true the provider not automatically run the internal cleanup logic. Instead , it expects an external scheduler (like a cron job or Google Cloud Scheduler) to handle the cleanup by running `hf-gce purgeMachines` manual or on schedule. Make sure to point to the correct configuration file before running the command. | `false`                                ...
```
Hi @clineberry, all requested changes have been fixed and pushed

- Renamed `SCHEDULER_USE_PURGE_CMD` -> `AUTO_RUN_TRIM_DB_CMD` with default state to `true`
- Changed `RETURN_VM_TTL` -> `RETURNED_VM_TTL`
- Updated `machines.py` logic
- Updated `SYMPHONY_GCE_INSTALL.md` doc

All changes pushed in commit 35a685e8caf2fcf38fb4e7ec95a6cba882d48b0d

clineberry reviewed (commented):
I approve these updates. There's still the need for @corykim to  sign off on the machines.py changes.

saalieri committed:
Add debounce to async remove_expired_returned_machines to prevent rapid repeated calls

clineberry reviewed (approved):
LGTM

saalieri commented on hf-provider/src/gce_provider/db/machines.py:
```diff
@@ -651,3 +663,62 @@ def _quick_check(self) -> Tuple[bool, Dict[str, Any]]:
+            selectQuery = f"""
+                SELECT machine_name
+                    FROM machines
+                WHERE machine_state IN ({MachineState.DELETED.value}, {MachineState.DELETE_REQUESTED.value})
```
I should have coordinated with you.

saalieri committed:
Remove auto-trim trigger from group delete handler and narrow cleanup query

saalieri committed:
Merge remote-tracking branch 'origin/main' into feat/35-db-trim

corykim reviewed (approved):
LG, thank you for the changes
